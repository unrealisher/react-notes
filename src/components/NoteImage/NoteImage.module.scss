.main {
  position: relative;
  z-index: 2;
  overflow: hidden;
  border-radius: 16px;
}

.image {
  margin-top: -2px;
  margin-left: -2px;
  width: calc(100% + 4px);
  height: auto;
  object-fit: cover;
}

.wrapper {
  height: 100%;
  display: flex;
  flex-direction: column;
  padding: 24px;
  padding-top: 12px;
}
