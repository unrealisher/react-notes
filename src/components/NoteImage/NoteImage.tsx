import React from "react";

import INote from "../../interfaces/INote";

import NoteFooter from "./../NoteFooter/NoteFooter";

import styles from "./NoteImage.module.scss";

interface IProps {
  note?: INote;
  tags?: string[] | undefined;
  setPatchItem: Function;
  setPopup: Function;
}

export const NoteImage = (props: IProps): JSX.Element => {
  const { note, tags, setPatchItem, setPopup } = props;
  return (
    <React.Fragment>
      {note && <img src={note.url} className={styles.image} alt="Картинка" />}
      {note && (
        <div className={styles.wrapper}>
          <NoteFooter
            note={note}
            tags={tags}
            setPatchItem={setPatchItem}
            setPopup={setPopup}
          />
        </div>
      )}
    </React.Fragment>
  );
};
export default NoteImage;
