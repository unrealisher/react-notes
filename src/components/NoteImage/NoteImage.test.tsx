import React from "react";
import { shallow } from "enzyme";
import { NoteImage } from "./NoteImage";

describe("NoteImage", () => {
  const note = {
    id: 7,
    type: "image",
    url:
      "https://avatars.mds.yandex.net/get-pdb/1534889/a869b7ef-32da-41e7-8703-326ae860e67f/orig",
    tags: [3],
    size: "m",
    created: 1548969900000
  };
  const tags = ["коты"];
  const setPatchItem = (): void => {};
  const setPopup = (): void => {};
  it("correct render", () => {
    const component = shallow(
      <NoteImage
        note={note}
        tags={tags}
        setPatchItem={setPatchItem}
        setPopup={setPopup}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
