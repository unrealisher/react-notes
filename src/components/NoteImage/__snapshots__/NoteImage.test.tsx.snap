// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteImage correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoteImage
    note={
      Object {
        "created": 1548969900000,
        "id": 7,
        "size": "m",
        "tags": Array [
          3,
        ],
        "type": "image",
        "url": "https://avatars.mds.yandex.net/get-pdb/1534889/a869b7ef-32da-41e7-8703-326ae860e67f/orig",
      }
    }
    setPatchItem={[Function]}
    setPopup={[Function]}
    tags={
      Array [
        "коты",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <img
          alt="Картинка"
          className="image"
          src="https://avatars.mds.yandex.net/get-pdb/1534889/a869b7ef-32da-41e7-8703-326ae860e67f/orig"
        />,
        <div
          className="wrapper"
        >
          <Connect(NoteFooter)
            note={
              Object {
                "created": 1548969900000,
                "id": 7,
                "size": "m",
                "tags": Array [
                  3,
                ],
                "type": "image",
                "url": "https://avatars.mds.yandex.net/get-pdb/1534889/a869b7ef-32da-41e7-8703-326ae860e67f/orig",
              }
            }
            setPatchItem={[Function]}
            setPopup={[Function]}
            tags={
              Array [
                "коты",
              ]
            }
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Картинка",
          "className": "image",
          "src": "https://avatars.mds.yandex.net/get-pdb/1534889/a869b7ef-32da-41e7-8703-326ae860e67f/orig",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(NoteFooter)
            note={
              Object {
                "created": 1548969900000,
                "id": 7,
                "size": "m",
                "tags": Array [
                  3,
                ],
                "type": "image",
                "url": "https://avatars.mds.yandex.net/get-pdb/1534889/a869b7ef-32da-41e7-8703-326ae860e67f/orig",
              }
            }
            setPatchItem={[Function]}
            setPopup={[Function]}
            tags={
              Array [
                "коты",
              ]
            }
          />,
          "className": "wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "note": Object {
              "created": 1548969900000,
              "id": 7,
              "size": "m",
              "tags": Array [
                3,
              ],
              "type": "image",
              "url": "https://avatars.mds.yandex.net/get-pdb/1534889/a869b7ef-32da-41e7-8703-326ae860e67f/orig",
            },
            "setPatchItem": [Function],
            "setPopup": [Function],
            "tags": Array [
              "коты",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <img
            alt="Картинка"
            className="image"
            src="https://avatars.mds.yandex.net/get-pdb/1534889/a869b7ef-32da-41e7-8703-326ae860e67f/orig"
          />,
          <div
            className="wrapper"
          >
            <Connect(NoteFooter)
              note={
                Object {
                  "created": 1548969900000,
                  "id": 7,
                  "size": "m",
                  "tags": Array [
                    3,
                  ],
                  "type": "image",
                  "url": "https://avatars.mds.yandex.net/get-pdb/1534889/a869b7ef-32da-41e7-8703-326ae860e67f/orig",
                }
              }
              setPatchItem={[Function]}
              setPopup={[Function]}
              tags={
                Array [
                  "коты",
                ]
              }
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Картинка",
            "className": "image",
            "src": "https://avatars.mds.yandex.net/get-pdb/1534889/a869b7ef-32da-41e7-8703-326ae860e67f/orig",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(NoteFooter)
              note={
                Object {
                  "created": 1548969900000,
                  "id": 7,
                  "size": "m",
                  "tags": Array [
                    3,
                  ],
                  "type": "image",
                  "url": "https://avatars.mds.yandex.net/get-pdb/1534889/a869b7ef-32da-41e7-8703-326ae860e67f/orig",
                }
              }
              setPatchItem={[Function]}
              setPopup={[Function]}
              tags={
                Array [
                  "коты",
                ]
              }
            />,
            "className": "wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "note": Object {
                "created": 1548969900000,
                "id": 7,
                "size": "m",
                "tags": Array [
                  3,
                ],
                "type": "image",
                "url": "https://avatars.mds.yandex.net/get-pdb/1534889/a869b7ef-32da-41e7-8703-326ae860e67f/orig",
              },
              "setPatchItem": [Function],
              "setPopup": [Function],
              "tags": Array [
                "коты",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
